react-app/src/components/projects/ProjectDetail.tsx

```diff
@@ -77,17 +101,31 @@ const ProjectDetail = (props: IProjectDetail) => {
     //Somewhat evil reduce where we collect information from the status and tasks lookup so that we can
     //get data for the status and task names to be displayed when we enumarete the tasks associated to the project itself
     //in the documentation history section.
-    return data?.parsedBody.Tasks.reduce((acc: Record<string, Array<any>>, curr) => {
-      const fullTask = tasks.find((a) => a.Id === curr.TaskId);
-      const fullStatus = statuses.find((a) => a.Id === fullTask.StatusId);
+    const reduceMap = data?.parsedBody.Tasks.reduce(
+      (acc: Record<string, IStatusHistoryStruct>, curr) => {
+        const fullTask = tasks.find((a) => a.Id === curr.TaskId);
+        const fullStatus = statuses.find((a) => a.Id === fullTask.StatusId);
+        if (!acc[fullStatus.Name]) {
```

Comment:
If fullStatus doesn't have a value (no status with that id found), then this causes a rendering error. I noticed it on Approved For Exemption status.

Reply:
Tasks should always have a status, unlike notes where I've left some null, so this could indicate a different issue. But I gave them both a fallback value regardless.